### PR TITLE
feat(ros-z-cli): add topic frequency estimation

### PR DESCRIPTION
**Pull request by schmidma**
## Overview

Adds `rosz hz` for topic frequency estimation. The command reports local receive-rate stats plus per-publisher source-rate stats, with compact text output and JSON-lines support.

## Motivation

`rosz echo` is useful for inspecting payloads, but frequency checks only need raw samples and timing metadata. This gives operators a quick `ros2 topic hz`-style check without requiring schema discovery or payload decoding.

## Implementation Guide

- `ros-z`: extends automatic dynamic subscriber discovery with `.raw()`, discovering publisher `TypeInfo` from graph metadata and building a raw subscriber without querying schema services.
- `ros-z-cli`: adds `hz` parsing, rolling interval stats, count/duration/continuous modes, compact text rendering, and JSON report objects.
- Tests cover parser validation, rolling stats, duration scheduling edge cases, and an e2e publisher with schema service disabled.

## Verification

- `cargo fmt --check`
- `cargo test --doc -p ros-z`
- `cargo nextest run -p ros-z`
- `cargo nextest run -p ros-z-cli`

### CODE REVIEW

**Comment by rmburg** [comment]
this has conflicts, please resolve

File: crates/ros-z-cli/src/commands/hz.rs [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+
+    #[cfg(unix)]
+    fn capture_stderr(mut f: impl FnMut()) -> String {
+        use std::{
+            fs::File,
+            io::{Read, Seek, SeekFrom, Write},
+            os::fd::{AsRawFd, FromRawFd, RawFd},
+        };
+
+        unsafe extern "C" {
```
Comment (oleflb) [inline crates/ros-z-cli/src/commands/hz.rs]: add test dependency to `libc` or even `nix` for safe bindings?

File: crates/ros-z-cli/src/model/hz.rs [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        let variance = self
+            .intervals
+            .iter()
+            .map(Duration::as_secs_f64)
+            .map(|seconds| {
+                let delta = seconds - mean;
+                delta * delta
+            })
+            .sum::<f64>()
+            / intervals as f64;
```
Comment (oleflb) [inline crates/ros-z-cli/src/model/hz.rs]: nit: this is biased, prefer (n-1) estimate

**Review by oleflb — CHANGES_REQUESTED** [review]
Looks good. I don't really care about findings. If you want we can merge now

File: crates/ros-z-cli/src/commands/hz.rs (hunk 1 above)
Comment (schmidma) [reply crates/ros-z-cli/src/commands/hz.rs]: removed the test

File: crates/ros-z-cli/src/model/hz.rs (hunk 2 above)
Comment (schmidma) [reply crates/ros-z-cli/src/model/hz.rs]: yes, done

**Review by oleflb — APPROVED** [review]
